### ISSUE TITLE
Error: Couldn't generate key

### ISSUE THREAD

**Issue by KenSharp** [bisq-network/bisq#1780]
When loading Bisq I receive a window that tells me to report this issue.

![bisq](https://user-images.githubusercontent.com/13607576/47095261-d0e31a00-d224-11e8-9d32-4be514a5ffe4.png)

From the log:

````
Oct-17 15:50:25.796 [checkCryptoThread] ERROR b.c.c.Encryption: JCE cannot authenticate the provider BC 
Oct-17 15:50:25.825 [checkCryptoThread] ERROR b.c.s.CommonSetup: Uncaught Exception from thread checkCryptoThread 
Oct-17 15:50:25.826 [checkCryptoThread] ERROR b.c.s.CommonSetup: throwableMessage= Couldn't generate key 
Oct-17 15:50:25.826 [checkCryptoThread] ERROR b.c.s.CommonSetup: throwableClass= class java.lang.RuntimeException 
Oct-17 15:50:25.848 [checkCryptoThread] ERROR b.c.s.CommonSetup: Stack trace:
java.lang.RuntimeException: Couldn't generate key
	at bisq.common.crypto.Encryption.generateSecretKey(Encryption.java:231)
	at bisq.network.crypto.EncryptionService.encryptHybridWithSignature(EncryptionService.java:112)
	at bisq.core.app.BisqSetup.lambda$checkCryptoSetup$4(BisqSetup.java:440)
	at java.lang.Thread.run(Thread.java:748)
````

[bisq.log](https://github.com/bisq-network/bisq/files/2487986/bisq.log)

I tried running Bisq with the Ubuntu Xenial provided packages of JRE and JDK 8 and 9 but it doesn't even get this far, so I'm using:

`$ /opt/jdk1.8.0_191/bin/java -Xss1280k -jar /opt/Bisq/app/Bisq-0.8.0.jar`

which seems to allow me to use the application without issue, except for this error message.

Does this mean I'm not secure?

**Comment by KenSharp** [comment]
Ugh. This is the wrong place, isn't it?

**Comment by ManfredKarrer** [comment]
If you run from the jar file you need to use jdk 10 as we upgraded to java 10 in last release.

**Comment by boyofbow** [comment]
I have same problem with windows 10
[bisq.log](https://github.com/bisq-network/bisq/files/2490835/bisq.log)

**Comment by ManfredKarrer** [comment]
Can you try to use lastest master version? You need to install openjdk10.

**Comment by KenSharp** [comment]
OpenJDK9 is the latest available on Xenial.
https://packages.ubuntu.com/search?suite=all&arch=i386&searchon=names&keywords=openjdk

I'll have to try again when I update to Bionic.

> If you run from the jar file you need to use jdk 10 as we upgraded to java 10 in last release.

What other way is there to run it?

````
$ /opt/Bisq/Bisq
Segmentation fault (core dumped)
$ /opt/Bisq/Bisq -BjvmOptions=-Xss1280k
Segmentation fault (core dumped)
````

**Comment by ManfredKarrer** [comment]
If you use the binary all should be fine.

**Comment by KenSharp** [comment]
The binary that segfaults?

**Comment by ghost** [comment]
I do not remember any ubuntu user reporting worries with the Bisq v0.8.0 binary.
Binaries are here : https://bisq.network/downloads/

**Comment by KenSharp** [comment]
That's where I got it from.

**Comment by freimair** [comment]
the issue that the JRE cannot create a private key should be solved with the upcoming 0.9 release as Bisq now uses the onboard JCE (and therefore, ```b.c.c.Encryption: JCE cannot authenticate the provider BC ``` should not happen anymore)

In the meantime, did you try openjdk?

**Comment by boyofbow** [comment]
No how do I do that

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by KenSharp** [comment]
> the issue that the JRE cannot create a private key should be solved with the upcoming 0.9 release

Glad to see someone found the issue.

> In the meantime, did you try openjdk?

Yup. Same result.
